### PR TITLE
[Improvement][API Test] Add API test cases for process definition controller

### LINKED ISSUE #13945: [Improvement][API Test] Add API test cases for ProcessDefinitionController
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

* Add API test cases for `ProcessDefinitionController` 
* part of #10411

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of the pull request

* This PR will close: #13945

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by EricGao888 — COMMENTED** [review]
Should remove `dolphinscheduler-ui/pnpm-lock.yaml`.

**Review by kezhenxu94 — DISMISSED** [review]

File: dolphinscheduler-ui/pnpm-lock.yaml [hunk 1]
```diff
@@ -3663,4 +3663,4 @@ packages:
     resolution: {integrity: sha512-DRUM4ZLnoaT0PBVvGBDO9oWIDBKFdAVieNWxWwK0niYzJCMwGchRk21/hsE+RKkIveH3XHCyvXcJDkgLVvfizQ==}
     dependencies:
       tslib: 2.3.0
-    dev: false
+    dev: false
```
Comment (SbloodyS) [inline dolphinscheduler-ui/pnpm-lock.yaml]: We should remove inrelevent change.

File: dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api/test/cases/ProcessDefinitionAPITest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import org.junit.jupiter.api.Test;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.util.EntityUtils;
+
+@DolphinScheduler(composeFiles = "docker/basic/docker-compose.yaml")
+@Slf4j
+public class ProcessDefinitionAPITest {
```
Comment (SbloodyS) [inline dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api/test/cases/ProcessDefinitionAPITest.java:50]: Please add this to github actions.

Comment (EricGao888) [reply dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api/test/cases/ProcessDefinitionAPITest.java:50]: Sure

File: dolphinscheduler-ui/pnpm-lock.yaml (hunk 1 above)
Comment (EricGao888) [reply dolphinscheduler-ui/pnpm-lock.yaml]: Fixed

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**